### PR TITLE
Handled case for _start replication with leader cluster at higher version than the follower cluster

### PR DESCRIPTION
**Pull request by saikaranam-amazon**
Signed-off-by: Sai Kumar <karanas@amazon.com>

### Description
Handled case for _start replication with leader cluster at higher version than the follower cluster
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/kotlin/org/opensearch/replication/util/ValidationUtil.kt [hunk 1]
```diff
@@ -73,4 +75,25 @@ object ValidationUtil {
...
+    fun validateLeaderIndexMetadata(leaderIndexMetadata: IndexMetadata) {
+        if(Version.CURRENT.before(leaderIndexMetadata.creationVersion)) {
+            val err = "Leader index[${leaderIndexMetadata.index.name}] is on " +
+                    "higher version [${leaderIndexMetadata.creationVersion}] than follower [${Version.CURRENT}]"
+            log.error(err)
+            throw IllegalArgumentException(err)
+        }
+        if(Version.CURRENT.before(leaderIndexMetadata.upgradedVersion)) {
+            val err = "Leader index[${leaderIndexMetadata.index.name}] is upgraded with " +
+                    "higher version [${leaderIndexMetadata.creationVersion}] than follower [${Version.CURRENT}]"
```
Comment (soosinha) [inline src/main/kotlin/org/opensearch/replication/util/ValidationUtil.kt:94]: log should probably be leaderIndexMetadata.upgradedVersion

Comment (saikaranam-amazon) [reply src/main/kotlin/org/opensearch/replication/util/ValidationUtil.kt]: Done

**Review by soosinha — APPROVED** [review]

File: src/main/kotlin/org/opensearch/replication/util/ValidationUtil.kt [hunk 2]
```diff
@@ -73,4 +75,25 @@ object ValidationUtil {
...
+            throw IllegalArgumentException(err)
+        }
+        if(Version.CURRENT.before(leaderIndexMetadata.upgradedVersion)) {
+            val err = "Leader index[${leaderIndexMetadata.index.name}] is upgraded with " +
+                    "higher version [${leaderIndexMetadata.upgradedVersion}] than follower [${Version.CURRENT}]"
+            log.error(err)
+            throw IllegalArgumentException(err)
+        }
+    }
+}
```
Comment (krishna-ggk) [inline src/main/kotlin/org/opensearch/replication/util/ValidationUtil.kt:99]: Can you elaborate the reasoning (and also include that in the code-comment)?

Is it due to potential new index settings that aren't recognized on follower?

Also, can you clarify if there is any additional handling required if leader is upgraded post replication start?

Comment (saikaranam-amazon) [reply src/main/kotlin/org/opensearch/replication/util/ValidationUtil.kt]: Yes, the restriction is due to settings etc being backward compatible in higher version of OpenSearch. Updated the code comment to capture this info.
I will update the documentation with the additional cases for cluster upgrade with replication in progress.

**Review by soosinha — APPROVED** [review]

**Review by ankitkala — APPROVED** [review]

**Pull request merged**